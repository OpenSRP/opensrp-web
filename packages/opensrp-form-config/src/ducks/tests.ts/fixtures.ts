export const fixManifestReleases = [
    {
        identifier: '1.0.11',
        json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:00:38.413+02:00',
        updatedAt: '2020-06-08T15:00:38.413+02:00',
    },
    {
        identifier: '1.0.12',
        json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:03:00.950+02:00',
        updatedAt: '2020-06-08T15:03:00.950+02:00',
    },
    {
        identifier: '1.0.16',
        json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.16',
        createdAt: '2020-06-09T14:40:04.189+02:00',
        updatedAt: '2020-06-09T14:40:04.189+02:00',
    },
];
