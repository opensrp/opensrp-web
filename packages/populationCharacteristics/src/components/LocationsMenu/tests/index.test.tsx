import React from 'react';
import { mount, shallow } from 'enzyme';
import _ from 'lodash';
import { LocationMenu } from '..';
import { locHierarchy } from '../../../ducks/locations/tests/fixtures';
import toJson from 'enzyme-to-json';

const props = {
    activeLocationId: '02ebbc84-5e29-4cd5-9b79-c594058923e9',
    isLast: true,
    loadLocsettings: jest.fn(),
    locationDetails: locHierarchy.locationsHierarchy.map['75af7700-a6f2-448c-a17d-816261a7749a'],
    popLocPopup: jest.fn(),
    showLocPopUp: '',
};

describe('components/LocationsMenu', () => {
    it('renders without crashing', () => {
        shallow(<LocationMenu {...props} />);
    });

    it('render menu correctly', () => {
        const wrapper = mount(<LocationMenu {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.find('.locations').simulate('click');
        expect(props.popLocPopup).toHaveBeenCalledWith(expect.any(Object), '75af7700-a6f2-448c-a17d-816261a7749a');
        wrapper.find('.popuptext div').simulate('click');
        expect(props.loadLocsettings).toHaveBeenCalledWith(expect.any(Object), '8400d475-3187-46e4-8980-7c6f0a243495');
        wrapper.unmount();
    });

    it('renders coreectly with popup id', () => {
        const newProps = {
            ...props,
            isLast: false,
            showLocPopUp: '02ebbc84-5e29-4cd5-9b79-c594058923e9',
        };
        const wrapper = mount(<LocationMenu {...newProps} />);

        expect(toJson(wrapper.find('.locations'))).toMatchSnapshot('location');
        expect(toJson(wrapper.find('.popup'))).toMatchSnapshot('popup');
        wrapper.find('.locations').simulate('click');
        expect(props.popLocPopup).toHaveBeenCalledWith(expect.any(Object), '75af7700-a6f2-448c-a17d-816261a7749a');
        wrapper.find('.popuptext div').simulate('click');
        expect(props.loadLocsettings).toHaveBeenCalledWith(expect.any(Object), '8400d475-3187-46e4-8980-7c6f0a243495');
        wrapper.unmount();
    });
});
