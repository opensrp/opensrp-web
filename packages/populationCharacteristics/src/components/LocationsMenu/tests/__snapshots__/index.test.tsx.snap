// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LocationsMenu render menu correctly 1`] = `
<LocationMenu
  activeLocationId="02ebbc84-5e29-4cd5-9b79-c594058923e9"
  isLast={true}
  loadLocsettings={[MockFunction]}
  locationDetails={
    Object {
      "children": Object {
        "8400d475-3187-46e4-8980-7c6f0a243495": Object {
          "children": Object {
            "f4d22dad-f211-4476-bdf0-09fb20f7f64f": Object {
              "id": "f4d22dad-f211-4476-bdf0-09fb20f7f64f",
              "label": "Outreach",
              "node": Object {
                "locationId": "f4d22dad-f211-4476-bdf0-09fb20f7f64f",
                "name": "Outreach",
                "parentLocation": Object {
                  "locationId": "8400d475-3187-46e4-8980-7c6f0a243495",
                  "name": "Lobi",
                  "parentLocation": Object {
                    "locationId": "75af7700-a6f2-448c-a17d-816261a7749a",
                    "name": "ME",
                    "voided": false,
                  },
                  "voided": false,
                },
                "voided": false,
              },
              "parent": "8400d475-3187-46e4-8980-7c6f0a243495",
            },
          },
          "id": "8400d475-3187-46e4-8980-7c6f0a243495",
          "label": "Lobi",
          "node": Object {
            "locationId": "8400d475-3187-46e4-8980-7c6f0a243495",
            "name": "Lobi",
            "parentLocation": Object {
              "locationId": "75af7700-a6f2-448c-a17d-816261a7749a",
              "name": "ME",
              "voided": false,
            },
            "voided": false,
          },
          "parent": "75af7700-a6f2-448c-a17d-816261a7749a",
        },
      },
      "id": "75af7700-a6f2-448c-a17d-816261a7749a",
      "label": "ME",
      "node": Object {
        "locationId": "75af7700-a6f2-448c-a17d-816261a7749a",
        "name": "ME",
        "tags": Array [
          "Country",
        ],
        "voided": false,
      },
    }
  }
  popLocPopup={[MockFunction]}
  showLocPopUp=""
>
  <div
    className="locations-menu"
    key="75af7700-a6f2-448c-a17d-816261a7749a"
  >
    <a
      className="locations "
      onClick={[Function]}
    >
      <span>
        ME
      </span>
      <span
        className="drop-down"
      />
    </a>
    <div
      className="popup"
    >
      <div
        className="popuptext loc-popup "
      >
        <div
          key="8400d475-3187-46e4-8980-7c6f0a243495"
          onClick={[Function]}
        >
          Lobi
        </div>
      </div>
    </div>
  </div>
</LocationMenu>
`;

exports[`components/LocationsMenu renders coreectly with popup id: location 1`] = `
<a
  className="locations "
  onClick={[Function]}
>
  <span>
    ME
  </span>
  <span
    className="drop-down"
  />
  /
</a>
`;

exports[`components/LocationsMenu renders coreectly with popup id: popup 1`] = `
<div
  className="popup"
>
  <div
    className="popuptext loc-popup "
  >
    <div
      key="8400d475-3187-46e4-8980-7c6f0a243495"
      onClick={[Function]}
    >
      Lobi
    </div>
  </div>
</div>
`;
